storage/src/adb/keyless.rs

```diff
@@ -296,13 +389,28 @@ impl<E: Storage + Clock + Metrics, V: Codec, H: CHasher> Keyless<E, V, H> {
         let section = self.current_section();
         let operation = Operation::Append(value);
         let encoded_operation = operation.encode();
-        let (offset, _) = self.log.append(section, operation).await?;
-        self.locations.append(offset).await?;
-        self.mmr
-            .add_batched(&mut self.hasher, &encoded_operation)
-            .await?;
 
+        // Create a future that appends the operation to the log and updates locations with the
+        // resulting offset.
+        let log_loc_fut = async {
+            let (offset, _) = self.log.append(section, operation).await?;
```

Comment:
in the future, I suppose we could precompute this offset to run all these writes concurrently.

Reply:
good idea